### PR TITLE
Filtering on belongs_to associations should sort by the name field by default

### PR DESCRIPTION
**Pull request by whitepaperclip**
Any thoughts on this? I was thinking through the possible scenarios for which this *wouldn't* be desirable behavior. The one that sticks out: a blogging platform where it would be helpful to see the most recent posts at the top of the filter select. However, this new default behavior isn't all that bad, since they could still find it alphabetically.

As of right now, if the association uses a UUID—which they should, in almost all cases—the items in the select box appear in a completely random order. (On Postgres, anyway.)

A potential remedy for this would be to allow passing sort options in the filter:

`filter [:posts, labels: [posts: "Posts"], sort_orders: [posts: [asc: :name]]]`

I thought about adding this, but I honestly think the filter options are best completely refactored to something less redundant:

`filter [posts: [label: "Posts", order_by: [asc: :name]]]`

Thoughts?

### CODE REVIEW

**Comment by smpallen99** [comment]
@whitepaperclip I like this idea of adding a filter option and refactoring the API too. your suggestion is much nicer.

**Comment by whitepaperclip** [comment]
Thanks! I'll start working on a refactor.

It probably won't be backwards-compatible. :(

**Comment by smpallen99** [comment]
Its time to create a new version anyway. So we can manage the breaking change.

**Comment by smpallen99** [comment]
Really appreciate your contributions. Didn't realize how far behind I was. Getting caught up...

**Comment by whitepaperclip** [comment]
No worries! Thanks!

File: lib/ex_admin/themes/admin_lte2/filter.ex [hunk 1]
```diff
@@ -121,7 +122,12 @@ defmodule ExAdmin.Theme.AdminLte2.Filter do
     id = "q_#{owner_key}"
     name_label = field_label(name, defn)
     repo = Application.get_env :ex_admin, :repo
-    resources = repo.all assoc
+    name_field = ExAdmin.Helpers.get_name_field(defn.resource_model)
```
Comment (sineed) [inline lib/ex_admin/themes/admin_lte2/filter.ex]: @whitepaperclip I guess there is a mistake. Here the association name field must return

Comment (whitepaperclip) [inline lib/ex_admin/themes/admin_lte2/filter.ex]: This has been updated in another pull request, which should be merged shortly.

Comment (tralamazza) [inline lib/ex_admin/themes/admin_lte2/filter.ex]: it's `assoc` and not `defn.resource_model`, the child table might not have that field name.

**Pull request merged**